apps/designer/app/designer/shared/sync/queue.ts

```diff
@@ -0,0 +1,67 @@
+import { atom, type WritableAtom } from "nanostores";
+
+// After this amount of retries without success, we consider connection status error.
+const MAX_RETRY_RECOVERY = 5;
+
+export type SyncStatus = "syncing" | "idle" | "recovering" | "error";
+
+type BaseResult = { ok: boolean };
+
+export const state: {
+  status: WritableAtom<SyncStatus>;
```

Comment:
status is not a state, it's state container. Maybe split separately since you anyway want to use it as export.